surefire-report-parser/src/main/java/org/apache/maven/plugins/surefire/report/TestSuiteXmlParser.java

```diff
@@ -171,6 +171,11 @@ public void startElement(String uri, String localName, String qName, Attributes
                         testCase.setError(attributes.getValue("message"), attributes.getValue("type"));
                         currentSuite.incrementNumberOfErrors();
                         break;
+                    case "system-out":
+                    case "system-err":
+                        currentElement = new StringBuilder();
+                        parseContent = true;
+                        break;
```

Comment:
This is incomplete/imprecise. Look at the schema: https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report.xsd. Testcase can have stdout/stderr, but also reruns and flakes. We should basically overwrite in that case. We need a way to make sure that these are direct descendants.

Reply:
My PR ist half a year old. I created it, because you asked me to. It is meant to be something you can commit on top of and refine. "Allow edits by maintainers" is active for this PR. Please do so.

Currently, my personal situation does not permit me to spend time on it, because on top of being busy with work I am in the process of relocating to another continent and was also thrown back by a recent stay in hospital after an accident, which cost me extra time. This is not an excuse, just an explanation. I am still interested in getting this issue resolved, only the time window to further collaborate on it has closed for now.